### PR TITLE
Allow the app mount element to be configurable.

### PR DESCRIPTION
**Pull request by jpickwell**
**This is based off of #28.**

The `appMountElement` option can be set to the name of an element; e.g. `'article'`. This element is used for all `appMountIds`. This could be expanded in the future to allow the element to be set per id, but that would be best implemented with breaking changes.

The `appMountIds` option, and thus the `appMountId` option, has been modified to be similar to the `links` and `scripts` options.

This enhancement was inspired by #24.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -20,29 +24,43 @@ $ npm install html-webpack-template --save-dev
...
+- `template`: Specify this module's `index.ejs` file.
 
 And some other optional:
 
-- `appMountId`: div element id on which you plan to mount a javascript app (can include multiple elements using the `appMountIds` array).
-- `devServer`: Insert the webpack-dev-server hot reload script at this host:port/path (eg, http://localhost:3000).
-- `baseHref`: Adjust the url for relative urls in the document ([MDN](https://developer.mozilla.org/en/docs/Web/HTML/Element/base)).
-- `filename`: The file to write the HTML to. Defaults to `index.ejs`.
-   You can specify a subdirectory here too (eg: `assets/admin.html`).
+- `appMountElement`: The HTML element name to use as the application element(s). Default value: `'div'`.
```
Comment (jaketrent) [inline README.md:32]: I don't feel a strong need for this.  I'm fine with only allowing a generic div w/in which to mount an app.

File: README.md [hunk 2]
```diff
@@ -20,29 +24,43 @@ $ npm install html-webpack-template --save-dev
...
+- `appMountId`: The application element id on which you plan to mount a JavaScript app. The value of this option is
+  prepended to the `appMountIds` option. See `appMountIds` for possible values.
+- `appMountIds`: An array of application element ids or objects.
```
Comment (jaketrent) [inline README.md:35]: This is a new feature that I don't have a great desire to add.  If a projects has a special case where they need many app mount points, they can implement their own template.  This does not feel like a common case feature.

File: README.md [hunk 3]
```diff
@@ -54,44 +72,52 @@ Here's an example webpack config illustrating how to use these options in your `
...
       baseHref: 'http://example.com/awesome',
-      devServer: 3001,
+      devServer: 'http://localhost:3001',
       googleAnalytics: {
         trackingId: 'UA-XXXX-XX',
         pageViewOnLoad: true
       },
       meta: {
-        description: "a better default template for html-webpack-plugin"
+        description: 'A better default template for html-webpack-plugin.'
```
Comment (jaketrent) [inline README.md:87]: I like the style of the old version.

File: README.md [hunk 4]
```diff
@@ -54,44 +72,52 @@ Here's an example webpack config illustrating how to use these options in your `
...
       template: require('html-webpack-template'),
-      //template: 'node_modules/html-webpack-template/index.ejs',
+      // template: 'node_modules/html-webpack-template/index.ejs',
 
       // Optional
+      appMountElement: 'article',
       appMountId: 'app',
       baseHref: 'http://example.com/awesome',
-      devServer: 3001,
+      devServer: 'http://localhost:3001',
```
Comment (jaketrent) [inline README.md:81]: Why this change to add localhost here?

File: index.ejs [hunk 5]
```diff
@@ -1,108 +1,106 @@
...
-
-  <title><%= htmlWebpackPlugin.options.title || 'Webpack App'%></title>
-
-  <% if (htmlWebpackPlugin.files.favicon) { %>
-  <link rel="shortcut icon" href="<%= htmlWebpackPlugin.files.favicon%>">
-  <% } %>
-  <% if (htmlWebpackPlugin.options.mobile) { %>
-  <meta name="viewport" content="width=device-width, initial-scale=1">
-  <% } %>
+<% var item, key %>
```
Comment (jaketrent) [inline index.ejs:1]: This change potentially makes the internals of the template easier to read by itself.  But it introduces a level of abstraction in the template that I kind of feel like avoiding.  I would like to come to this template and see immediately how it works.  If there are if statements and template writes that map straight back to webpack values, there is little to understand.  If there is now an intermediate variables that rename values assigned to them, I have to understand that layer as well.  I'd prefer to leave this out.

**Review by jaketrent — CHANGES_REQUESTED** [review]
Thanks for the work here.  

I still feel like this PR is not splitting things up well enough to focus on one issue at a time.  I feel like I'd appreciate a couple separately-merged PRs, like: 
- Improving all the docs, fixing grammar and style, adding defaults
- Refactoring the template, introducing vars (but I don't recommend we pursue this specific one)
- Adding new feature X (eg, appMountElement)
- Adding new feature Y (eg, appMountIds) (though also not recommended)

Even so, I've tried to give a bit of feedback on the wealth of changes given here.

File: README.md (hunk 4 above)
Comment (jpickwell) [reply README.md:81]: A port number by itself is interpreted as a relative path to a directory with that name. This also makes the example line-up with the documentation of the property.

File: README.md (hunk 2 above)
Comment (jpickwell) [reply README.md:35]: I didn't add the appMountIds option. You added it in 92ff11423494000b287a0eb6f2a56012183bbd37.

**Comment by jpickwell** [comment]
I will split everything up. I made this pull request before I saw your comment on #28.

File: README.md (hunk 2 above)
Comment (jaketrent) [reply README.md:35]: lol!  sorry for the misread.  I'll have to go talk to myself about this. :)

File: README.md (hunk 4 above)
Comment (jaketrent) [reply README.md:81]: oh, I see, this is just a documentation change.

**Comment by jaketrent** [comment]
Closing in favor of separate merges.
